app/src/main/java/com/cmput301f20t21/bookfriends/ui/login/LoginActivity.java

```diff
@@ -0,0 +1,127 @@
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+        usernameLayout = findViewById(R.id.login_username_layout);
+        usernameField = findViewById(R.id.login_username_field);
+        usernameField.addTextChangedListener(new AfterTextChangedWatcher() {
+            @Override
+            public void afterTextChanged(Editable usernameString) {
+                String username = usernameString.toString();
+                checkUsername(username); // return value not needed
+            }
+        });
+        passwordLayout = findViewById(R.id.login_password_layout);
+        passwordField = findViewById(R.id.login_password_field);
+        passwordField.addTextChangedListener(new AfterTextChangedWatcher() {
+            @Override
+            public void afterTextChanged(Editable passwordString) {
+                String password = passwordString.toString();
+                checkPassword(password); // return value not needed
+            }
+        });
+    }
+
+    /**
+     * check the username field for emptiness, adding error message if field is empty
+     * @param username - the username that the user entered
+     * @return true if the username is not empty, false if username is empty
+     */
+    private boolean checkUsername(String username) {
+        boolean isUsernameEmpty = username.isEmpty();
+        if (isUsernameEmpty) {
+            usernameLayout.setError(getString(R.string.username_empty_error));
+        } else {
+            usernameLayout.setError(null);
+        }
+        return !isUsernameEmpty;
+    }
+
+    /**
+     * check the username field for emptiness, adding error message if field is empty
+     * @param password - the password that the user entered
+     * @return true if the password is not empty, false if password is empty
+     */
+    private boolean checkPassword(String password) {
+        boolean isPasswordEmpty = password.isEmpty();
+        if (isPasswordEmpty) {
+            passwordLayout.setError(getString(R.string.password_empty_error));
+        } else {
+            passwordLayout.setError(null);
+        }
+        return !isPasswordEmpty;
+    }
+
+    /**
+     * TODO: fill out javadoc comments once finished
+     * @param view
+     */
+    public void onCreateAccountClicked(View view) {
+        // start the activity here, no result is needed
+        // created user will be stored directly to firestore
+    }
+
+    /**
+     * method called when user clicked the "login" button
+     * the function will validate the inputs and authenticate the user
+     * user will be redirected to the main screen if authentication is successful
+     * @param view - the view associated with the button
+     */
+    public void onLoginClicked(View view) {
+        Editable usernameText = usernameField.getText();
```

Comment:
if usernameField = usernameLayout.getEditText(), I don't think usernameText will ever be null. Don't think we need to check it.

Reply:
If we want to be really sure, we could check if getEditText() exists.

Reply:
yeah before if I just used getText().toString() it gives a possible NPE warning, using getEditText().getText().toString() the warning disappeared, removed the check